### PR TITLE
Better better quoting

### LINKED ISSUE #1195: Better quoter addon can quote any and all text on page, should include bbcode as well.
**Describe the bug**

When using the better-quoter addon, you can select any text on the page and then press any quote button, and the selected text will be quoted. In addition (this should probably be a separate issue), this addon should include the correct bbcode of the quoted text, as scratch's native quoter does. There should also be an optional option to automatically close bbcode tags when quoting, but again, that's another issue.

**To Reproduce**

1. Go to any non-closed forum topic
2. select any amount of text from anywhere on the page
3. click any quote button on the page
4. you will see that all of the selected text will be quoted, and there will be no bbcode from any posts you selected.

**Expected behavior**

It should either not do anything if you try to quote things outside of a post, just quote the original post or just quote the part that is selected inside the post.
The selected post's bbcode should be included in the quote.

**Screenshots**

**Device (please complete the following information):**

- OS: windows 10
- Browser: chrome 87
- Version: sa 1.7.0-prerelease

**Additional context**

I should be able to fix this (if someone could assign me that would be nice). But there are a couple of options, either one of those as detailed in expected behaviour, or we could do what discourse does and add a little quote button next to the end of the selected text if it's all in one post, and use that to quote small parts, and the usual button to quote as normal. This makes the most sense to me.
Regarding the bbcode, we can fetch https://scratch.mit.edu/post/POST_ID/source to get the bbcode of it, and then figure out where about the person has selected it and then go from there... tbh that's not really viable but oh well. 
This will require a complete revamp of this addon.

### PR DESCRIPTION
**Pull request by pufferfish101007**
**Resolves**

Resolves #1195

**Changes**

When using the better-quoting addon, quoting now also includes bbcode (other than scratchblocks), and adds me to the credits

**Reason for changes**

They are useful and without it this addon is pretty useless (for me anyway)

**Tests**

Tested on chrome. However, there may be some tags/scenarios that I haven't thought of and therefore cause bugs.

**additional info**
- scratchblocks aren't supported
- it doesn't use innerHTML or outerHTML anymore
- you can still quote across quote, siggie, author info and date. Again, I'm too lazy to fix it.
- pls merge soon because I forgot to make a new branch so I'll have to make another fork again :)

### CODE REVIEW

File: addons/better-quoter/addon.json [hunk 1]
```diff
@@ -14,5 +18,7 @@
     }
   ],
   "tags": ["forums", "community"],
-  "l10n": true
+  "l10n": true,
+  "warning": "Scratchblocks tags are not supported, and the bbcode may not be 100% accurate. To get fully accurate quotes, quote as normal and delete any parts that you don't need."
```
Comment (TheColaber) [inline addons/better-quoter/addon.json:22]: this was updated, use the system according to https://github.com/ScratchAddons/ScratchAddons/pull/957#issue-533087250

File: addons/better-quoter/addon.json [hunk 2]
```diff
@@ -5,6 +5,10 @@
     {
       "name": "TheColaber",
       "link": "https://scratch.mit.edu/users/TheColaber"
+    },
+    {
+      "name": "pufferfish101007",
+      "link": "https://scratch.mit.edu"
```
Comment (TheColaber) [inline addons/better-quoter/addon.json:11]: link to your profile

File: addons/better-quoter/addon.json (hunk 1 above)
Comment (TheColaber) [reply addons/better-quoter/addon.json:22]: and don't do warning, do notice.

**Comment by GrahamSH-LLK** [comment]
Instead of using innerhtml etc. to figure out the tags, can't you use the API (or scratchdb)?

**Comment by RedGuy12** [comment]
@mxmou suggested this because it would be faster
________________________________

File: addons/better-quoter/addon.json (hunk 2 above)
Comment (pufferfish101007) [reply addons/better-quoter/addon.json:11]: Woops thanks for letting me know

**Comment by pufferfish101007** [comment]
> Instead of using innerhtml etc. to figure out the tags, can't you use the API (or scratchdb)?

Yes as @RedGuy12 said, this is a lot faster, and also it would be practically impossible to actually tell where the selection is in comparison to the source. So to answer your question, no. Unless we want slowness.

**Comment by pufferfish101007** [comment]
Actually @GrahamSH-LLK I just realised that the innerHTML's only there for browser compatibility- I'll try removing it and see what happens, and while I'm at it I'll get rid of the outerhtmls.

**Comment by WorldLanguages** [comment]
Moving to v1.8 milestone so that translators can focus on the new extension (not addon) locals, such as the new "popups" category and the update notification.

**Comment by WorldLanguages** [comment]
Is this OK to merge?

**Comment by pufferfish101007** [comment]
> Is this ok to merge?

Yes, it's all good to go.

**Pull request merged**